cdap-docs/admin-manual/source/security/authorization.rst

```diff
@@ -304,90 +303,155 @@ Streams
+privileges for :ref:`Integrations: Apache Sentry <apache-sentry>`.
+
+You can use the :ref:`CDAP CLI <cdap-cli>` to issue :ref:`security commands <cli-available-commands-security>`.
+Wildcard can be used to grant or revoke actions on multiple entities byincluding ``*`` and ``?`` in the entity name:
 
 - To grant a principal privileges to perform certain actions on an entity, use::
 
     > grant actions <actions> on entity <entity-id> to <principal-type> <principal-name>
+    > revoke actions <actions> on entity <entity-id> from <principal-type> <principal-name>
 
   where:
 
   - ``<actions>`` is a comma-separated list of privileges, any of *READ, WRITE, EXECUTE,* or *ADMIN*.
 
   - ``<entity>`` is of the form ``<entity-type>:<entity-id>``, where ``<entity-type>`` is
-    one of ``namespace``, ``artifact``, ``app``, ``dataset``, ``program``, ``stream``, or ``view``.
+    one of ``namespace``, ``artifact``, ``application``, ``dataset``, ``program``, ``stream``, ``dataset_type`` or
+    ``dataset_module``.
 
   - For namespaces, ``<entity-id>`` is composed from the namespace, such as
     ``namespace:<namespace-name>``.
 
-  - For artifacts and apps, ``<entity-id>`` is composed of the namespace, entity name, and
-    version, such as ``<namespace-name>.<artifact-name>.<artifact-version>`` or
-    ``<namespace-name>.<app-name>.<app-version>``.
+  - For datasets, streams, artifacts and apps, ``<entity-id>`` is the namespace and entity names, such as
+    ``<namespace-name>.<dataset-name>``, ``<namespace-name>.<stream-name>``, ``<namespace-name>.<artifact-name>``,
+    and ``<namespace-name>.<app-name>``.
 
   - For programs, ``<entity-id>`` includes the application name and the program type:
     ``<namespace-name>.<app-name>.<program-type>.<program-name>``. ``<program-type>`` is
     one of flow, mapreduce, service, spark, worker, or workflow.
 
-  - For datasets and streams, ``<entity-id>`` is the namespace and entity names, such as
-    ``<namespace-name>.<dataset-name>`` or ``<namespace-name>.<stream-name>``.
+  - For datasets, streams, artifacts and apps, ``<entity-id>`` is the namespace and entity names, such as
+    ``<namespace-name>.<dataset-name>``, ``<namespace-name>.<stream-name>``, ``<namespace-name>.<artifact-name>``,
+    and ``<namespace-name>.<app-name>``.
+
+  - ``<principal-type>`` can **only** be ``role`` since Sentry only supports grants privileges to roles.
+
+  - Wildcard can be used in each entity name to grant privileges to multiple entities. For example,
+    ``namespace:ns*`` represents all namespaces that starts with ``ns``.
+    ``namespace:ns?`` represents all namespaces that starts with ``ns`` and follows by a single character.
+    ``program:ns1.app1.*`` represents all types of programs in application app1 in namespace ns1.
+
+- To add the role to other principal, use::
+
+    > add role <role-name> to <principal-type> <principal-name>
```

Comment:
role can only be granted to group we should document that

Reply:
I see. But I tested our CLI, it is working for user. Will add the restriction in the docs.